src/node_url.cc

```diff
@@ -1283,7 +1283,9 @@ void URL::Parse(const char* input,
         }
         break;
       case kNoScheme:
-        cannot_be_base = base->flags & URL_FLAGS_CANNOT_BE_BASE;
+        if (has_base) {
```

Comment:
+1 to splitting it out. It may be worthwhile having a test case for it also since it doesn't seem to be caught by the WPT tests at all.